llap-common/src/java/org/apache/hadoop/hive/llap/LlapThreadLocalStatistics.java

```diff
@@ -0,0 +1,176 @@
+   * Input list:
+   * 1. FileSystem.Statistics (scheme: file)
+   * 2. FileSystem.Statistics (scheme: hdfs)
+   * 3. FileSystem.Statistics (scheme: hdfs)
+   * Output map:
+   * file : LlapThreadLocalStatistics.StatisticsData (1)
+   * hdfs : LlapThreadLocalStatistics.StatisticsData (2 + 3)
+   */
+  public LlapThreadLocalStatistics(ThreadMXBean mxBean, List<FileSystem.Statistics> allStatistics) {
+    cpuTime = mxBean == null ? -1 : mxBean.getCurrentThreadCpuTime();
+    userTime = mxBean == null ? -1 : mxBean.getCurrentThreadUserTime();
+
+    for (FileSystem.Statistics statistics : allStatistics) {
+      schemeToThreadLocalStats.merge(statistics.getScheme(),
+          new StatisticsData(statistics.getThreadStatistics()),
+          (statsCurrent, statsNew) -> statsCurrent.merge(statistics.getThreadStatistics()));
+    }
+  }
+
+  // This method iterates on the other LlapThreadLocalStatistics's schemes, and subtract them from this one if it's
+  // present here too.
+  public LlapThreadLocalStatistics subtract(LlapThreadLocalStatistics other) {
+    for (Map.Entry<String, LlapThreadLocalStatistics.StatisticsData> otherStats :
+        other.schemeToThreadLocalStats.entrySet()){
+      schemeToThreadLocalStats.computeIfPresent(otherStats.getKey(),
+          (thisScheme, stats) -> stats.subtract(otherStats.getValue()));
+    }
+
+    cpuTime -= other.cpuTime;
+    userTime -= other.userTime;
+
+    return this;
+  }
+
+  public void fill(TezCounters tezCounters) {
+    for (Map.Entry<String, LlapThreadLocalStatistics.StatisticsData> threadLocalStats :
+        schemeToThreadLocalStats.entrySet()){
+      String scheme = threadLocalStats.getKey();
+      StatisticsData stats = threadLocalStats.getValue();
+      tezCounters.findCounter(scheme, FileSystemCounter.BYTES_READ).increment(stats.bytesRead);
+      tezCounters.findCounter(scheme, FileSystemCounter.BYTES_WRITTEN).increment(stats.bytesWritten);
+      tezCounters.findCounter(scheme, FileSystemCounter.READ_OPS).increment(stats.readOps);
+      tezCounters.findCounter(scheme, FileSystemCounter.LARGE_READ_OPS).increment(stats.largeReadOps);
+      tezCounters.findCounter(scheme, FileSystemCounter.WRITE_OPS).increment(stats.writeOps);
+    }
+
+    if (cpuTime >= 0 && userTime >= 0) {
+      tezCounters.findCounter(LlapThreadLocalStatistics.LlapExecutorCounters.EXECUTOR_CPU_NS).increment(cpuTime);
+      tezCounters.findCounter(LlapThreadLocalStatistics.LlapExecutorCounters.EXECUTOR_USER_NS).increment(userTime);
+    }
+  }
+
+  public String toString(){
+    return String.format("LlapThreadLocalStatistics: %s", schemeToThreadLocalStats.toString());
+  }
+
+  /**
+   * Convenience class over Hadoop's FileSystem.Statistics.StatisticsData.
+   * Unfortunately, neither the fields, nor the convenience methods (e.g. StatisticsData.add, StatisticsData.negate)
+   * are available here as they are package protected, so we cannot reuse them.
+   */
+  public static class StatisticsData {
```

Comment:
I would suggest to rename it which distinct it from FileSystem.Statistics.StatisticsData.
Like HiveStatisticsData or LlapStatisticsData.

Reply:
makes sense, I would pick LlapStatisticsData or LlapFileSystemStatisticsData

Reply:
Your choice. ;)

Reply:
renamed to LlapFileSystemStatisticsData, resolving this conversation